### PR TITLE
Frogger game

### PR DESCRIPTION
**Pull request by YuliiaHordiichuk**
### Frogger game

[Demo](https://yuliiahordiichuk.github.io/frogger/)
[Code base](https://github.com/YuliiaHordiichuk/frogger)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/YuliiaHordiichuk/frogger/app.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    }
+}; 
+
+Enemy.prototype.checkCollision = function() {
+    if(Math.abs(player.x - this.x) < 80 && this.y === player.y) {
+        setTimeout(() => {
+            alert ('You lost'); 
+            player.restart();
+        },0) 
+    };
```
Comment (OleksiyRudenko) [inline submissions/YuliiaHordiichuk/frogger/app.js:40]: Classes shouldn't refer to any global variables directly.
Exclusions are global constants and, in this particular exercise, entities provided by the engine (`Resources`).

File: submissions/YuliiaHordiichuk/frogger/app.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    player.handleInput(allowedKeys[e.keyCode]);
+});
+
+const getEnemyY = (row) => startY - row * stepY; 
+
+const allEnemies = [
+    new Enemy(-150, getEnemyY(2)), 
+    new Enemy (-300, getEnemyY(3)), 
+    new Enemy (-50, getEnemyY(4))
+];
```
Comment (OleksiyRudenko) [inline submissions/YuliiaHordiichuk/frogger/app.js:126]: There is a pattern here. Let's practise `Array` methods to make this frgament dryer.
Imagine, you need create dozens of enemies.
Consider the example below:
`[[1, 5], [12, 17]].map(([x, y]) => new Circle(x ,y, constRadius))`
or
`[1, 12].map((x, index) => new Circle(x, index + 2, constRadius))`

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@YuliiaHordiichuk a rally good job!
One fix to foolow OOP principles and one pattern optimization and we're done.

File: submissions/YuliiaHordiichuk/frogger/app.js (hunk 1 above)
Comment (YuliiaHordiichuk) [reply submissions/YuliiaHordiichuk/frogger/app.js]: thanks for the comments, done

File: submissions/YuliiaHordiichuk/frogger/app.js (hunk 2 above)
Comment (YuliiaHordiichuk) [reply submissions/YuliiaHordiichuk/frogger/app.js]: done

**Review by OleksiyRudenko — APPROVED** [review]
@YuliiaHordiichuk great job!

**Pull request merged**